vispy/plot/plotwidget.py

```diff
@@ -260,10 +266,11 @@ def mesh(self, vertices=None, faces=None, vertex_colors=None,
                 raise ValueError('vertices, faces, and fname must be None if '
                                  'fname is not None')
         else:
-            meshdata = MeshData(vertices, faces)
+            meshdata = MeshData(vertices, faces, face_colors=face_colors,
+                                vertex_colors=vertex_colors)
```

Comment:
For some reason, face and vertex colors must be set in MeshData, or it doesn't work. I'm not sure if scene.Mesh just isn't applying the keywords properly, or if there is some other interaction.